modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -402,12 +402,8 @@ public IgniteFutureImpl<Void> start(
                 return;
             }
 
-            if (!reqGrpIds.isEmpty()) {
-                finishProcess(fut0.rqId, new IllegalArgumentException(OP_REJECT_MSG + "Cache group(s) was not " +
-                    "found in the snapshot [groups=" + reqGrpIds.values() + ", snapshot=" + snpName + ']'));
-
-                return;
-            }
+            assert reqGrpIds.isEmpty() : "Cache group(s) was not found in the snapshot [groups=" + reqGrpIds.values()
```

Comment:
Why this change from exception to assert?

Reply:
> Why this change from exception to assert?

Because this duplicates the check within MetadataVerificationJob#checkMeta(SnapshotMetadata meta) and must not occure here.